### PR TITLE
Mention procmacros in the intro docs and other minor doc tweaks

### PR DESCRIPTION
**Pull request by mhammond**

### CODE REVIEW

File: docs/manual/src/index.md [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
-Note that this tool will not help you ship a Rust library to these platforms, but simply not have to write bindings code by hand. [Related](https://i.kym-cdn.com/photos/images/newsfeed/000/572/078/d6d.jpg).
+Note that this tool will not help you ship a Rust library to these platforms, but it will help you avoid writing bindings code by hand.
+[Related](https://i.kym-cdn.com/photos/images/newsfeed/000/572/078/d6d.jpg).
 
 ## Design
 
-UniFFI requires to write an Interface Definition Language (based on [WebIDL](https://heycam.github.io/webidl/)) file describing the methods and data structures available to the targeted languages.
-This .udl (UniFFI Definition Language) file, whose definitions must match with the exposed Rust code, is then used to generate Rust *scaffolding* code and foreign-languages *bindings*. This process can take place either during the build process or be manually initiated by the developer.
+UniFFI requires you to describe your interface via either proc-macros or in an Interface Definition Language (based on [WebIDL](https://heycam.github.io/webidl/)) file.
```
Comment (badboy) [inline docs/manual/src/index.md:11]: This redirects now. Should we just use the final URL?

```suggestion
UniFFI requires you to describe your interface via either proc-macros or in an Interface Definition Language (based on [WebIDL](https://webidl.spec.whatwg.org/)) file.
```

**Review by badboy — CHANGES_REQUESTED** [review]

File: docs/manual/src/Motivation.md [hunk 2]
```diff
@@ -28,11 +28,10 @@ continues to evolve.
...
 
 * Implement your software component as a `cdylib` crate in Rust; let's say the code is in `./src/lib.rs`.
-* Specify the desired component API using an *Interface Definition Language* (specifically, a variant of WebIDL) in a separate file like `./src/lib.udl`.
-* Run `uniffi-bindgen scaffolding ./src/lib.udl` to generate a bunch of boilerplate rust code that exposes this API as a C-compatible FFI layer,
-  and include it as part of your crate.
+* Optionally, describe parts of your component API using proc-macros directly in `lib.rs`.
+* Optionally, describe parts of your component API using an *Interface Definition Language* in a separate file like `./src/lib.udl`, you'll generate a bunch of boilerplate rust code that exposes this API as a C-compatible FFI layer, and include it as part of your crate.
 * `cargo build` your crate as normal to produce a shared library.
-* Run `uniffi-bindgen generate ./src/lib.udl -l kotlin` to generate a Kotlin library that can load your shared library
+* Run `uniffi-bindgen generate ... -l kotlin` to generate a Kotlin library that can load your shared library
```
[the remark this answers is not in the document]
Comment (mhammond) [reply docs/manual/src/Motivation.md:34]: It indicates omitted args. I believe ellipses are commonly used to indicate that items were omitted, but happy to change this to something else if you disagree about that.

I removed the reference to the udl because I was trying to avoid the impression that UDL files are required - if you aren't using a UDL file, you don't specify it - or to put it another way, the example was factually wrong in certain scenarios. Instead, you'd probably specify the name of a shared library - but the example doesn't specify what that is, and I feel adding that would distract from the content here.

In other words, I intentionally omitted these details because I felt that accurately covering all possible arguments is not relevant at this point of the manual. Could you please detail the exact change you'd like to see?

[the remark this answers is not in the document]
Comment (badboy) [reply docs/manual/src/Motivation.md:34]: But that's my problem: it's unclear what it is omitting here.
But I won't be blocking on this, and maybe in the motivation that's good enough.

**Review by badboy — APPROVED** [review]

File: docs/manual/src/Motivation.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mhammond) [reply docs/manual/src/Motivation.md:34]: OK - I added a little note about the omitted args with a link to the relevant docs
